client/src/quic_client.rs

```diff
@@ -301,6 +302,35 @@ impl QuicClient {
                 }
             }
         };
+
+        let new_stats = connection.connection.stats();
```

Comment:
Should we be concerned the stats are loaded too frequently compared with previous implementation only during get_connection?

Reply:
I think we should measure if this adversely impacts sending transaction performances. If we can get some metrics, it would be great

Reply:
The way the quic client is used, with the previous logic, every time send_buffer is called, get_connection would have been called anyway (since we use get_connection) to get the quic connection to call into. This is also the intended usage paradigm, so I don't think this is a concern.

Reply:
Getting some stats now. But remember, _send_buffer is only called once per batch send. After the first send in the batch, all the other sends use send_buffer_using_conn.

Reply:
Right!

Reply:
Looks just simple Atomic operations.